addons/enemybar/bars.lua

```diff
@@ -0,0 +1,284 @@
+
+-- Meta class
+bars = {x_res = windower.get_windower_settings().x_res,y_res = windower.get_windower_settings().y_res}
+
+-- Base class method new
+
+function bars.new(o, bar_settings)
+   o = o or {}
+   o.width = bar_settings.width 
+   o.color = bar_settings.color 
+   o.font = bar_settings.font
+   o.font_size = bar_settings.font_size
+   o.show_dist = bar_settings.show_dist
+   o.show_target = bar_settings.show_target
+   o.show_target_icon = bar_settings.show_target_icon
+   o.show_action = bar_settings.show_action
+   o.show_debuff = bar_settings.show_debuff
+   bars.initialize(o)
+   bars.move(o, bar_settings.pos.x, bar_settings.pos.y)
+   return o
+end
+
+function bars.destroy(o)
+	if not o then return end
```

Comment:
You check for the existence of `o`, despite it never being `nil`, and also checking again at the call site. You can pretty much remove every one of these `if` checks in this module. It is good practice to do this for public facing libraries to validate unknown input (although I personally do not always agree with that either), but for internal libraries you always control the input, so it is not necessary.

Reply:
It actually is called when o could be nil. it's an edge case though. I put it in here to clean up the code in enemybar.lua, since it's used in more places.